protos/datamodel/costs.proto

```diff
@@ -0,0 +1,236 @@
+syntax = "proto3";
+
+import "datamodel/common.proto";
+
+package datamodel.v2;
+option go_package = "github.com/opencost/opencost/core/pkg/model/pb;pb";
+
+// CostMetrics represents different cost accounting methods
+// Mirrors OpenCost's multiple cost calculation approaches
+message CostMetrics {
+  // List cost (raw cost without discounts)
+  float listCost = 1;
+
+  // Net cost (list cost minus discounts)
+  float netCost = 2;
+
+  // Amortized net cost (includes reserved instance amortization)
+  float amortizedNetCost = 3;
+
+  // Invoiced cost (actual billed amount)
+  float invoicedCost = 4;
+
+  // Amortized cost (list cost with RI amortization)
+  float amortizedCost = 5;
+
+  // Percentage of cost attributed to Kubernetes workloads
+  float kubernetesPercent = 6;
+}
+
+// ResourceCost represents cost for a specific resource type
+// Follows OpenCost's base cost + adjustment pattern
+message ResourceCost {
```

Comment:
@Sean-Holcomb perhaps you and I should discuss this one.

I'm wondering if we, rather, just want something like

```proto
enum Unit {
  UNIT_CPU_MILLICORE_HOUR = 1;
  UNIT_RAM_MIB_HOUR = 2;
  UNIT_GPU_HOUR = 3;
  UNIT_GB_HOUR = 4;
  UNIT_BYTE = 5;
}

message ResourceQuantity {
  Unit unit = 1;
  float quantity = 2;
}

message ResourcePrice {
  Unit unit = 1;
  float price = 2;
}
```

Then `Node` can have `ResourceQuantity` and `ResourcePrice` for CPU/RAM/GPU. And `Container` can have `ResourceQuantity` for CPU/RAM/GPU, but no price because it will reference a `Node` which has the pricing.